account_invoice_margin/models/account_move_line.py

```diff
@@ -48,7 +48,7 @@ def _get_purchase_price(self):
         self.ensure_one()
         return self.product_id.standard_price
 
-    @api.depends("product_id", "product_uom_id")
+    @api.depends("product_id", "product_uom_id", "move_id.move_type")
```

Comment:
Is it possible to change the type of an invoice "on the fly"?

Reply:
In odoo standard: action_switch_invoice_into_refund_credit_note
In any case, better to have the field dependencies. Even if you don't expect them to change, some other module might need it ^^